app/controllers/application_controller.rb

```diff
@@ -262,7 +270,9 @@ def check_student_consent
   # a hidden param :after_sign_in_path to the sign in form.
   def after_sign_in_path_for(resource)
     redirect_path = root_path
-    if current_user.portal_student
+    if ENV['RESEARCHER_REPORT_ONLY']
+      redirect_path = learner_report_path
```

Comment:
Yeah, with this we don't need the after_sign_in_path in the report link from the main portal. It will just ignore the after_sign_in_path if there is one set.  I don't think that will break anything though. The two places it is used in the link from the main portal, and when a anonymous user isn't authorized to access a page. In either case when it is a report_only portal, it seems reasonable to always try to send them to the learner_report_path.